src/components/Subscription/PublicCalendarSubscriptionPicker.vue

```diff
@@ -55,18 +62,38 @@ import { mapGetters } from 'vuex'
 import { findAllSubscriptions } from '../../services/caldavService.js'
 import holidayCalendars from '../../resources/holiday_calendars.json'
 import { uidToHexColor } from '../../utils/color.js'
+import { loadState } from '@nextcloud/initial-state'
 
 export default {
-	name: 'HolidaySubscriptionPicker',
+	name: 'PublicCalendarSubscriptionPicker',
 	components: {
 		NcButton,
 		NcModal,
 	},
+	props: {
+		showHolidays: Boolean,
+	},
 	data() {
-		const calendars = holidayCalendars.map(calendar => ({
-			...calendar,
-			source: 'https://www.thunderbird.net/media/caldata/' + calendar.filename,
-		}))
+		let calendars = []
+		if (this.showHolidays) {
+			calendars = holidayCalendars.map(calendar => ({
+				...calendar,
+				displayName: t('calendar', 'Holidays in {region}', {
+					region: calendar.country,
+				}),
+				name: calendar.country,
+				description: calendar.datespan,
+				source: 'https://www.thunderbird.net/media/caldata/' + calendar.filename,
+			}))
+		} else {
+			try {
+				const state = loadState('calendar', 'publicCalendars')
+				calendars = JSON.parse(state)
```

Comment:
This should also validate the parsed content to make sure each element has valid `displayName`, `name`, `description` and more importantly `source` attributes.

Reply:
With what result if one or more are invalid?

Reply:
You could simply filter the invalid elements out of the list.